gradio/components/imageslider.py

```diff
@@ -0,0 +1,292 @@
+"""gr.ImageSlider() component."""
+
+from __future__ import annotations
+
+import warnings
+from collections.abc import Callable, Sequence
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Literal, cast
+from urllib.parse import quote
+
+import numpy as np
+import PIL.Image
+from gradio_client import handle_file
+from gradio_client.documentation import document
+from PIL import ImageOps
+
+from gradio import image_utils
+from gradio.components.base import Component
+from gradio.data_classes import GradioRootModel, ImageData
+from gradio.events import Events
+from gradio.exceptions import Error
+
+
+class SliderData(GradioRootModel):
+    root: tuple[ImageData | None, ImageData | None] | None
+
+
+image_tuple = tuple[
+    str | PIL.Image.Image | np.ndarray | None, str | PIL.Image.Image | np.ndarray | None
+]
+
+
+if TYPE_CHECKING:
+    from gradio.components import Timer
+
+PIL.Image.init()  # fixes https://github.com/gradio-app/gradio/issues/2843
+
+
+@document()
+class ImageSlider(Component):
+    """
+    Creates an image component that can be used to upload images (as an input) or display images (as an output).
+
+    Demos: imageslider
+    """
+
+    EVENTS = [
+        Events.clear,
+        Events.change,
+        Events.stream,
+        Events.select,
+        Events.upload,
+        Events.input,
+    ]
+
+    data_model = SliderData
+
+    def __init__(
+        self,
+        value: image_tuple | Callable | None = None,
+        *,
+        format: str = "webp",
+        height: int | str | None = None,
+        width: int | str | None = None,
+        image_mode: (
+            Literal[
+                "1", "L", "P", "RGB", "RGBA", "CMYK", "YCbCr", "LAB", "HSV", "I", "F"
+            ]
+            | None
+        ) = "RGB",
+        type: Literal["numpy", "pil", "filepath"] = "numpy",
+        label: str | None = None,
+        every: Timer | float | None = None,
+        inputs: Component | Sequence[Component] | set[Component] | None = None,
+        show_label: bool | None = None,
+        show_download_button: bool = True,
+        container: bool = True,
+        scale: int | None = None,
+        min_width: int = 160,
+        interactive: bool | None = None,
+        visible: bool = True,
+        elem_id: str | None = None,
+        elem_classes: list[str] | str | None = None,
+        render: bool = True,
+        key: int | str | None = None,
+        show_fullscreen_button: bool = True,
+        position: float = 0.5,
```

Comment:
I'd suggest renaming for clarity:
```suggestion
        slider_position: float = 0.5,
```